src/zino/trapobservers/link_traps.py

```diff
@@ -0,0 +1,181 @@
+            # TODO: Record new number of flaps in event
+            # TODO: When flapcount modulo 100 is zero, log a message with flapping stats
+            pass
+        else:
+            event: PortStateEvent = self.state.events.get_or_create_event(device.name, port.ifindex, PortStateEvent)
+
+            event.portstate = new_state
+            event.port = port.ifdescr
+            event.ifindex = port.ifindex
+            port.state = new_state
+            if polldev := self.polldevs.get(device.name):
+                event.polladdr = polldev.address
+                event.priority = polldev.priority
+            event.descr = port.ifalias  # or value received from trap? see ldescr from legacy Zino
+
+            # TODO: If there is internal flapping state, log it in the event and clear internal state
+            # TODO: Set final flapcount in event
+
+            msg = f'{device.name}: intf "{port.ifdescr}" ix {port.ifindex} link{new_state.capitalize()}'
+            if reason:
+                event.reason = reason
+                msg = f'{msg}, "{reason}"'
+            _logger.info(msg)
+            event.add_log(msg)
+            self.state.events.commit(event)
+
+            # TODO: Poll single interface immediately to verify state change
+            # TODO: Schedule another single interface poll in two minutes
+
+    def is_port_ignored_by_patterns(self, device: DeviceState, ifdescr: str) -> bool:
+        if watch_pattern := self.get_watch_pattern(device):
+            if not re.match(watch_pattern, ifdescr):
+                return True
+
+        if ignore_pattern := self.get_ignore_pattern(device):
+            if re.match(ignore_pattern, ifdescr):
+                return True
+
+        return False
+
+    def is_port_ignored_by_policy(self, device: DeviceState, port: Port, is_up: bool) -> bool:
+        """Verifies that a link trap should be ignored, according to internal policies.
+
+        As commented in the original Zino 1 implementation:
+
+        > Should we ignore this trap message?
+        > If there's an open case for this router port, no.
+        > If received right after reload, yes.
+        > If state reported is the same as the one we have recorded,
+        > and a new trap does not reoccur within 5 minutes, yes.
+        > Otherwise, don't ignore the trap message.
+        """
+        now = zino.time.now()
+
+        # If there is an open case for router/port, do not ignore
+        if self.state.events.get(device.name, subindex=port.ifindex, event_class=PortStateEvent):
+            return False
+
+        # If we don't know that the device restarted recently, do not ignore
+        if not device.boot_time:
+            _logger.info("Oops! Do not know restart time for %s", device.name)
+            return False
+
+        if not port.state:
+            port.state = InterfaceState.UNKNOWN
+
+        if now - device.boot_time > timedelta(minutes=5):
```

Comment:
This magic 5 minutes could be a static value since its used a couple places and might want to configure in the future

Reply:
I assume you mean it could be a *constant*... fixed (although I'm not sure what to call this constant)